### PR TITLE
fix(es/compat): Update private field for bigint

### LINKED ISSUE #3700: Update a private field which is a bigint would cause type error
### Describe the bug

I see this issue in esbuild issue list, and in terms of private fields, esbuild does generate better code than babel/tsc/swc. However changes behavior to be like esbuild would need to change thousands of lines of test.

### Input code

```typescript
class A {
  #a = 123n
  foo() {
    this.#a++
  }
}
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.2.143&code=H4sIAAAAAAAAA0vOSSwuVnBUqOZSUFBOVLBVMDQyzgOy0%2FLzNTTBogoKJRmZxXrKidraQF4tVy0AkmwfZjMAAAA%3D&config=H4sIAAAAAAAAA0WOSw6DMAxEr1J5zYJ2QSXW3XIIKxhKlZ9sUxWh3L0hgNhZkzcvs8JHDLQrRGQh3i5ZvOIPWtAlkhieokIFKjlSninlm9HLENhtuKURzfIiExg18A5V0J9BR4o9KkI7oJVSRx5J8wckj%2FreZLkNQegAKnCTn4Zlc5vgIpPI9YR%2BtCeZssuFft6CtawtzgbS5Th6k3QHuK8j%2Fy2dsiTrwbw5OLo9a0jpD%2BI2wVYUAQAA

### Expected behavior

Run normally

### Actual behavior

Throw type error

### Version

1.2.144

### Additional context

_No response_

### LINKED ISSUE #4493: `classStaticPrivateFieldDestructureSet` get wrong result
### Describe the bug

-

### Input code

```typescript
class A {
    static #a = 1;
    foo() {
        ({ x: A.#a } = { x: 3 });
        console.log(A.#a);
    }
}

new A().foo();
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript"
    },
    "target": "es2021"
  },
  "module": {
    "type": "es6"
  },
  "isModule": true
}
```

### Playground link

[Playground](https://play.swc.rs/?version=1.2.174&code=H4sIAAAAAAAAA0vOSSwuVnBUqOZSAILiksSSzGQF5UQFWwVDa7BQWn6%2BhiZUGgQ0qhUqrBQc9YBqaoGqwDxjhVpNa7iK5Py84vycVL2c%2FHQNkDqoVC1XLRdXXmq5gqOGph7YVGsAHhcEHn0AAAA%3D&config=H4sIAAAAAAAAA6tWyipOVrKqVipILCpOLQKxiivzShIrlKyUUpNzE4uTizILSpRqdZRKEovSU0tAwsVGBkaGIKHc%2FJTSnFSQnpLKglSwlBlIPLPYFypTUlSaWgsA11yoYGQAAAA%3D)

### Expected behavior

print 3

### Actual behavior

Uncaught TypeError: Private static access of wrong provenance

### Version

1.2.174

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #3700
- fix #4493

### CODE REVIEW

**Comment by magic-akari** [comment]
I prefer the clean solution of esbuild as well.
But for this issue it can be solved simply first.

**Review by kdy1 — COMMENTED** [review]
This seems like a global regression of output size, even for users who does not use bigint values...

**Comment by magic-akari** [comment]
I added a `classPrivateFieldUpdate` method that does the similar trick as esbuild's `__privateWrapper` did.

**Comment by kdy1** [comment]
Cool!
I'll review it soon

File: packages/swc-helpers/src/_class_private_field_update.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import classExtractFieldDescriptor from "./_class_extract_field_descriptor";
```
Comment (kdy1) [inline packages/swc-helpers/src/_class_private_field_update.js:1]: Is this naming intentioal?

Comment (magic-akari) [reply packages/swc-helpers/src/_class_private_field_update.js:1]: Do you mean the prefix `_` from `_classApplyDescriptorUpdate`?
it's a typo.

**Review by kdy1 — APPROVED** [review]
Thank you!


---

swc-bump:
 - swc_ecma_transforms_base
 - swc_ecma_transforms_compat

**Pull request merged**